shared/hsm.py

```diff
@@ -177,7 +179,12 @@ def __init__(self, from_dict):
 
     def to_json(self):
         flds = [ 'mode', 'allow_zeroval_outs' ]
-        return dict((f, getattr(self, f, None)) for f in flds)
+        rv = OrderedDict()
+        for f in flds:
+            val = getattr(self, f, None)
+            if val:
```

Comment:
this would ignore both `None` and `False` --> does not seem right ignoring values